addons/website/models/ir_ui_view.py

```diff
@@ -273,8 +273,11 @@ def get_related_views(self, key, bundles=False):
         # website_id. (It will then always fallback on a website, this
         # method should never be called in a generic context, even for
         # tests)
-        self = self.with_context(website_id=self.env['website'].get_current_website().id)
-        return super(View, self).get_related_views(key, bundles=bundles)
+        current_website = self.env['website'].get_current_website()
+        self = self.with_context(website_id=current_website.id)
+        return super().get_related_views(key, bundles=bundles).with_context(
+            lang=current_website.default_lang_id._get_cached('code') if current_website else None,
```

Comment:
About `_get_cached`, not that this is perfectly fine to not use it in business code where we don't care about micro optimizing perfs and (especially) where the sql request to the website will be performed anyway.

Reaching this code, it's most likely that the query to the website table had to be done sooner. And if not, it's not that big of a deal considering that the `/website/get_switchable_related_views` route seems to be performing 272 sql requests